File: LeetCode/26.Remove_Duplicates_from_Sorted_Array/Mario/26.Remove_Duplicates_from_Sorted_Array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int pre = nums[0];
        int j = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != pre) {
                pre = nums[i];
                nums[j] = pre;
                j += 1;
            }
        }

        return j;
    }
}

File: LeetCode/26.Remove_Duplicates_from_Sorted_Array/26.Remove_Duplicates_from_Sorted_Array.swift
class Solution {
    func removeDuplicates(_ nums: inout [Int]) -> Int {
        // two pointers
        // i: tracks the position for unique elment
        // j: traverses entire nums
        let n = nums.count
        var i = 0
        for j in 1 ..< n {
            if nums[j] != nums[i] {
                i += 1
                nums[i] = nums[j]
            }
        }
        return i + 1
    }
}
